test/makeCompleteAction.test.js

```diff
@@ -161,4 +161,84 @@ describe("makeCompleteAction", () => {
       });
     });
   });
+
+  describe("with customSuccessPayload option", () => {
+    const customSuccessPayload = payload => payload;
+
+    it("should return a request action with a custom payload", () => {
+      const action = makeCompleteAction("Products", {
+        customSuccessPayload
+      });
+
+      expect(action("request", { test: 123 })).toEqual({
+        type: "GET_PRODUCTS",
+        test: 123
+      });
+    });
+
+    it("should return a success action with a custom payload", () => {
+      const action = makeCompleteAction("Products", {
+        customSuccessPayload,
+        payload: { test: 123 }
+      });
+
+      expect(action("success", { test: 123 })).toEqual({
+        type: "GET_PRODUCTS_SUCCESSFUL",
+        test: 123
+      });
+    });
+
+    it("should return a fail action with a regular payload", () => {
+      const action = makeCompleteAction("products", {
+        customSuccessPayload,
+        payload: { test: 123 }
+      });
+
+      expect(action("fail", { test: 123 })).toEqual({
```

Comment:
I don't understand why you're using a `customSuccessPayload` in an "fail" action here.

I think that you can create only three `describe` for the custom payloads:

1. `with customRequestPayload option`: testing the customRequestPayload in a request `action`;
1. `with customSuccessPayload option`: testing the customSuccessPayload in a success `action`;
1. `with customFailurePayload option`: testing the customFailurePayload in a fail `action`;

Makes sense?

Reply:
Made sense, done